src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts

```diff
@@ -396,8 +396,10 @@ export class XtermTerminal extends DisposableStore implements IXtermTerminal {
 
 	clearBuffer(): void {
```

Comment:
I notice now this is used by tasks and something wrong likely happens with and without this change:

https://github.com/microsoft/vscode/blob/9ade6a663662c5e4805038a6e3b3b4ba2f648472/src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts#L1297-L1299

Reply:
In tasks it may end up adding a decoration when presentation options say clear.